modules/social_features/social_album/social_album.module

```diff
@@ -562,3 +563,24 @@ function social_album_social_core_compatible_content_forms_alter(&$compatible_co
   $compatible_content_type_forms[] = 'node_album_form';
   $compatible_content_type_forms[] = 'node_album_edit_form';
 }
+
+/**
+ * Implements hook_node_links_alter().
+ */
+function social_album_node_links_alter(array &$links, NodeInterface $node, array &$context): void {
+  // 'Edit' button on current Album page for album creator and administrator.
+  if ($context['view_mode'] === 'full' && $node->getType() === 'album') {
+    $album_id = $node->id();
+    $current_user = \Drupal::currentUser();
+    $album_creator_id = $node->getOwnerId();
+    $link_edit_album = [
+      'title' => 'Edit',
+      'url' => Url::fromRoute('entity.node.edit_form', ['node' => $album_id]),
+      'weight' => '0',
+    ];
+
+    if ($current_user->hasPermission('edit own album content') && $current_user->id() === $album_creator_id) {
```

Comment:
We really should prevent checking permissions in UI code. Our intention here is:

```suggestion
    if ($album->access('edit')) {
```

That'll automatically invoke the access control handler and render the edit link for any user that can edit, regardless of how they got that permission (e.g. some future module adding collaboration). It'll also automatically pull in the current user by leaving the second parameter empty :)

Reply:
@Kingdutch such a solution already was proposed by @ribel and it doesn't work

Reply:
@ribel Do we know _why_ it doesn't work? 🙃 Because this not working sounds like a bug.

Reply:
I don't know why, but I am going to check this locally as well

Reply:
Probably the problem was that we should check access for `update` and not `edit` operation 🙈 

I have changed it to `$node->access('update', $account)` and after it all works correctly.